src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerPushImageFunctionalTest.groovy

```diff
@@ -0,0 +1,40 @@
+package com.bmuschko.gradle.docker.tasks.image
+
+import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
+import org.gradle.testkit.runner.BuildResult
+
+class DockerPushImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
+
+    def "fails on push error"() {
```

Comment:
Can you add a task for add a task for removing the container image after the push. You will also need to add a pull task to ensure that the container image is available before use. It's possible that your test case runs first and the image may not exist yet.

```
task removeImage(type: DockerRemoveImage) {
     dependsOn buildImage
     mustRunAfter pushImage, pullImage
     targetImageId buildImage.getImageId()
     force = true
}
```

Reply:
👍